cmd/sonobuoy/app/gen.go

```diff
@@ -22,21 +22,50 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 	"k8s.io/client-go/kubernetes"
 
 	ops "github.com/heptio/sonobuoy/pkg/client"
 	"github.com/heptio/sonobuoy/pkg/errlog"
 )
 
-var genopts ops.GenConfig
-
-var genFlags struct {
+type genFlags struct {
 	sonobuoyConfig SonobuoyConfig
 	mode           ops.Mode
 	rbacMode       RBACMode
 	kubecfg        Kubeconfig
 }
 
+var genopts ops.GenConfig
+var genflags genFlags
+
+func (g *genFlags) AddFlags(flags *pflag.FlagSet, cfg *ops.GenConfig, rbac RBACMode) {
+	genset := pflag.NewFlagSet("generate", pflag.ExitOnError)
+	AddModeFlag(&g.mode, genset)
+	AddSonobuoyConfigFlag(&g.sonobuoyConfig, genset)
+	AddKubeconfigFlag(&g.kubecfg, genset)
+	AddE2EConfigFlags(genset)
+	AddRBACModeFlags(&g.rbacMode, genset, rbac)
+	flags.AddFlagSet(genset)
+
+	AddNamespaceFlag(&cfg.Namespace, flags)
+	AddSonobuoyImage(&cfg.Image, flags)
+}
+
+func (g *genFlags) FillConfig(cmd *cobra.Command, cfg *ops.GenConfig) error {
```

Comment:
would it work if this returned a `*ops.GenConfig` directly instead of relying on a global var that we pass in?

Maybe there's a way to do this without having to pass in `*cobra.Command` -- it looks like that's only used for the e2e flags. Those look like special cases, do they need to be?

Reply:
> would it work if this returned a *ops.GenConfig directly instead of relying on a global var that we pass in?

It's passed in so it works with embedded structs

> Maybe there's a way to do this without having to pass in *cobra.Command -- it looks like that's only used for the e2e flags. Those look like special cases, do they need to be?
They do :( We need access to `Changed()` and we have to get that through `cmd`

Reply:
Okay, I'm gonna see if I can work around the latter case.